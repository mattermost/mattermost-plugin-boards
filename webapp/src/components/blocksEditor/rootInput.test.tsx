// Copyright (c) 2020-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.


import React from 'react'
import {render, screen, fireEvent} from '@testing-library/react'

import RootInput from './rootInput'

describe('components/blocksEditor/rootInput', () => {
    test('should match Display snapshot', async () => {
        const {container} = render(
            <RootInput
                onChange={jest.fn()}
                value='test-value'
                onChangeType={jest.fn()}
                onSave={jest.fn()}
            />,
        )
        expect(container).toMatchSnapshot()
    })

    test('should match Input snapshot', async () => {
        const {container} = render(
            <RootInput
                onChange={jest.fn()}
                value='test-value'
                onChangeType={jest.fn()}
                onSave={jest.fn()}
            />,
        )
        expect(container).toMatchSnapshot()
    })

    test('should match Input snapshot with menu open', async () => {
        const {container} = render(
            <RootInput
                onChange={jest.fn()}
                value=''
                onChangeType={jest.fn()}
                onSave={jest.fn()}
            />,
        )
        const input = screen.getByDisplayValue('')
        fireEvent.change(input, {target: {value: '/'}})
        expect(container).toMatchSnapshot()
    })

    test('should emit onChange event', async () => {
        const onChange = jest.fn()
        render(
            <RootInput
                onChange={onChange}
                value='test-value'
                onChangeType={jest.fn()}
                onSave={jest.fn()}
            />,
        )

        expect(onChange).not.toBeCalled()

        const input = screen.getByDisplayValue('test-value')
        fireEvent.change(input, {target: {value: 'test-value-'}})
        expect(onChange).toBeCalled()
    })

    test('should not emit onChangeType event when value is not empty and hit backspace', async () => {
        const onChangeType = jest.fn()
        render(
            <RootInput
                onChange={jest.fn()}
                value='test-value'
                onChangeType={onChangeType}
                onSave={jest.fn()}
            />,
        )

        expect(onChangeType).not.toBeCalled()
        const input = screen.getByDisplayValue('test-value')
        fireEvent.keyDown(input, {key: 'Backspace'})
        expect(onChangeType).not.toBeCalled()
    })

    test('should emit onSave event hit enter', async () => {
        const onSave = jest.fn()
        render(
            <RootInput
                onChange={jest.fn()}
                value='test-value'
                onChangeType={jest.fn()}
                onSave={onSave}
            />,
        )

        expect(onSave).not.toBeCalled()
        const input = screen.getByDisplayValue('test-value')
        fireEvent.keyDown(input, {key: 'Enter'})
        expect(onSave).toBeCalled()
    })

    test('should emit onChangeType event on menu option selected', async () => {
        const onChangeType = jest.fn()
        render(
            <RootInput
                onChange={jest.fn()}
                value=''
                onChangeType={onChangeType}
                onSave={jest.fn()}
            />,
        )

        const input = screen.getByDisplayValue('')
        fireEvent.change(input, {target: {value: '/'}})

        const option = screen.getByText('/title Creates a new Title block.')
        fireEvent.click(option)

        expect(onChangeType).toBeCalledWith(expect.objectContaining({name: 'h1'}))
    })
})
