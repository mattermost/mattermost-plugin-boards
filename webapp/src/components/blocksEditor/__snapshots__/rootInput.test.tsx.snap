// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/blocksEditor/rootInput should match Display snapshot 1`] = `
<div>
  <div
    class="RootInput css-b62m3t-container"
  >
    <span
      class="css-1f43avz-a11yText-A11yText"
      id="react-select-2-live-region"
    />
    <span
      aria-atomic="false"
      aria-live="polite"
      aria-relevant="additions text"
      class="css-1f43avz-a11yText-A11yText"
    />
    <div
      class=" css-1haocjs-control"
    >
      <div
        class=" css-b2z5qd-ValueContainer"
      >
        <div
          class=" css-1qgh1u0-Input"
          data-value="test-value"
        >
          <input
            aria-autocomplete="list"
            aria-describedby="react-select-2-placeholder"
            aria-expanded="false"
            aria-haspopup="true"
            autocapitalize="none"
            autocomplete="off"
            autocorrect="off"
            class=""
            id="react-select-2-input"
            role="combobox"
            spellcheck="false"
            style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
            tabindex="0"
            type="text"
            value="test-value"
          />
        </div>
      </div>
      <div
        class=" css-1hb7zxy-IndicatorsContainer"
      />
    </div>
  </div>
</div>
`;

exports[`components/blocksEditor/rootInput should match Input snapshot 1`] = `
<div>
  <div
    class="RootInput css-b62m3t-container"
  >
    <span
      class="css-1f43avz-a11yText-A11yText"
      id="react-select-3-live-region"
    />
    <span
      aria-atomic="false"
      aria-live="polite"
      aria-relevant="additions text"
      class="css-1f43avz-a11yText-A11yText"
    />
    <div
      class=" css-1haocjs-control"
    >
      <div
        class=" css-b2z5qd-ValueContainer"
      >
        <div
          class=" css-1qgh1u0-Input"
          data-value="test-value"
        >
          <input
            aria-autocomplete="list"
            aria-describedby="react-select-3-placeholder"
            aria-expanded="false"
            aria-haspopup="true"
            autocapitalize="none"
            autocomplete="off"
            autocorrect="off"
            class=""
            id="react-select-3-input"
            role="combobox"
            spellcheck="false"
            style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
            tabindex="0"
            type="text"
            value="test-value"
          />
        </div>
      </div>
      <div
        class=" css-1hb7zxy-IndicatorsContainer"
      />
    </div>
  </div>
</div>
`;

exports[`components/blocksEditor/rootInput should match Input snapshot with menu open 1`] = `
<div>
  <div
    class="RootInput css-b62m3t-container"
  >
    <span
      class="css-1f43avz-a11yText-A11yText"
      id="react-select-4-live-region"
    />
    <span
      aria-atomic="false"
      aria-live="polite"
      aria-relevant="additions text"
      class="css-1f43avz-a11yText-A11yText"
    />
    <div
      class=" css-1haocjs-control"
    >
      <div
        class=" css-b2z5qd-ValueContainer"
      >
        <div
          class=" css-1jqq78o-placeholder"
          id="react-select-4-placeholder"
        >
          Introduce your text or your slash command
        </div>
        <div
          class=" css-26cneq-Input"
          data-value=""
        >
          <input
            aria-autocomplete="list"
            aria-controls="react-select-4-listbox"
            aria-describedby="react-select-4-placeholder"
            aria-expanded="true"
            aria-haspopup="true"
            aria-owns="react-select-4-listbox"
            autocapitalize="none"
            autocomplete="off"
            autocorrect="off"
            class=""
            id="react-select-4-input"
            role="combobox"
            spellcheck="false"
            style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
            tabindex="0"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class=" css-1hb7zxy-IndicatorsContainer"
      />
    </div>
    <div
      class=" css-u0i6pk-MenuPortal"
    >
      <div
        class=" css-45h7mv-menu"
        id="react-select-4-listbox"
      >
        <div
          class=" css-1d1qzc4-MenuList"
        >
          <div
            aria-disabled="false"
            class=" css-8e5kjb-option"
            id="react-select-4-option-0"
            tabindex="-1"
          >
            /title Creates a new Title block.
          </div>
          <div
            aria-disabled="false"
            class=" css-x3yilo-option"
            id="react-select-4-option-1"
            tabindex="-1"
          >
            /subtitle Creates a new Sub title block.
          </div>
          <div
            aria-disabled="false"
            class=" css-x3yilo-option"
            id="react-select-4-option-2"
            tabindex="-1"
          >
            /subsubtitle Creates a new Sub Sub title block.
          </div>
          <div
            aria-disabled="false"
            class=" css-x3yilo-option"
            id="react-select-4-option-3"
            tabindex="-1"
          >
            /image Creates a new Image block.
          </div>
          <div
            aria-disabled="false"
            class=" css-x3yilo-option"
            id="react-select-4-option-4"
            tabindex="-1"
          >
            /text Creates a new Text block.
          </div>
          <div
            aria-disabled="false"
            class=" css-x3yilo-option"
            id="react-select-4-option-5"
            tabindex="-1"
          >
            /divider Creates a new Divider block.
          </div>
          <div
            aria-disabled="false"
            class=" css-x3yilo-option"
            id="react-select-4-option-6"
            tabindex="-1"
          >
            /list-item Creates a new List item block.
          </div>
          <div
            aria-disabled="false"
            class=" css-x3yilo-option"
            id="react-select-4-option-7"
            tabindex="-1"
          >
            /attachment Creates a new Attachment block.
          </div>
          <div
            aria-disabled="false"
            class=" css-x3yilo-option"
            id="react-select-4-option-8"
            tabindex="-1"
          >
            /quote Creates a new Quote block.
          </div>
          <div
            aria-disabled="false"
            class=" css-x3yilo-option"
            id="react-select-4-option-9"
            tabindex="-1"
          >
            /video Creates a new Video block.
          </div>
          <div
            aria-disabled="false"
            class=" css-x3yilo-option"
            id="react-select-4-option-10"
            tabindex="-1"
          >
            /checkbox Creates a new Checkbox block.
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
